src/renderers/shared/ReactDebugTool.js

```diff
@@ -52,6 +52,9 @@ var currentTimerStartTime = null;
 var currentTimerNestedFlushDuration = null;
 var currentTimerType = null;
 
+var beginLifeCycleTimerHasWarned = false;
```

Comment:
Just wanted to be safe about it. Thought maybe tests can produce situations that both functions generate warnings and then check the output of both. Changed it as you suggested in new commit.